Add docs warning about reload and debug flags (#949) (#1045) (#1047)
Kludex/uvicorn: docs/deployment.md

However, this style only works if you are not using multiprocessing (`workers=NUM`)
or reloading (`reload=True`), so we recommend using the import string style.

> Also note if you are using reloading(`reload=True`), you should put `uvicorn.run` into `if __name__ == '__main__'` clause in the main module to protect your "entry point".

Reviewer: ```suggestion
Also note if you are using either `reload=True`, either `debug=True`, you should put `uvicorn.run` into `if __name__ == '__main__'` clause in the main module".
```

Author: I haven't seen `debug` option in docs, maybe it's an outdated option? I just noticed `workers` option also need to attend at this position. It will produce same error message when it's not wrapped by a if-name-main clause.

Reviewer: > I haven't seen `debug` option in docs, maybe it's an outdated option? I just noticed `workers` option also need to attend at this position. It will produce same error message when it's not wrapped by a if-name-main clause.

whaaat ! https://github.com/encode/uvicorn/blob/master/uvicorn/main.py#L69-L71

mmm wondering why we dont see it in usage docs....

Reviewer: ok it's because it's a hidden click flag so wont be shown in `uvicorn --help`

let's keep the message for both, cant remember why i's hidden

Reviewer: ok debug was added in 546a60496d886f78d9f4dc60bd0f4aa15484f706 and was always a hidden command, let's keep it that way
